File: Java/Tabel.java
/*
Saya Mia Karisma Haq NIM [2102165] mengerjakan soal Latihan Praktikum-2 dalam mata kuliah DPBO
untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin
*/

// Kelas Tabel merupakan kelas untuk menampilkan tabel data list sivitasAkademik
public class Tabel {
    // Atribut pada kelas tabel terdiri dari baris dan kolom
    private int baris;
    private int kolom;

    // Constructor
    Tabel() 
    {

    }

    // Constructor : Overloading
    Tabel(int baris, int kolom)
    {
        this.baris = baris;
        this.kolom = kolom;
    }

    // Method set setiap atribut pada kelas tabel
    public void setBaris(int baris)
    {
        this.baris = baris;
    }

    public void setKolom(int kolom)
    {
        this.kolom = kolom;
    }

    // Method get setiap atribut pada kelas tabel
    public int getBaris()
    {
        return baris;
    }

    public int getKolom()
    {
        return kolom;
    }

    // Method untuk menampilkan list dalam bentuk tabel
    public void buatBaris(int baris, int kolom, String[][] data)
    {
        int i = 0, j = 0;
        int max[] = new int[kolom];
        for (i = 0; i < kolom; i++) {
            max[i] = 0;
            for (j = 0; j < baris; j++) {
                if (data[j][i].length() > max[i]) {
                    max[i] = data[j][i].length();
                }
            }
        }

        for (i = 0; i < baris; i++) {
            for (j = 0; j < kolom; j++) {
                for (int k = 0; k < max[j]; k++) {
                    System.out.print("-");
                }
                System.out.print("--");
            }
            System.out.println("---------");
            for (j = 0; j < kolom; j++) {
                System.out.print("|" + data[i][j]);
                for (int k = 0; k < max[j] - data[i][j].length(); k++) {
                    System.out.print(" ");
                }
                System.out.print("  ");
            }
            System.out.println("|");
        }
        for (j = 0; j < kolom; j++) {
            for (int k = 0; k < max[j]; k++) {
                System.out.print("-");
            }
            System.out.print("--");
        }
        System.out.println("---------");
    }
}

File: PHP/Program/Tabel.php
<!-- 
    Saya Mia Karisma Haq NIM [2102165] mengerjakan soal Latihan Praktikum-2 dalam mata kuliah DPBO
    untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin 
-->

<?php
    // Kelas Tabel merupakan kelas untuk menampilkan tabel data list Mahasiswa
    class Tabel
    {
        // Atribut yang digunakan pada kelas Tabel
        private $baris = 0;
        private $kolom = 0;
        
        // Constructor
        public function __construct($baris = 0, $kolom = 0)
        {
            $this->baris = $baris;
            $this->kolom = $kolom;
        }

        // Method setter untuk set setiap nilai atribut pada kelas Tabel
        public function setBaris($baris)
        {
            $this->baris = $baris;
        }

        public function setKolom($kolom)
        {
            $this->kolom = $kolom;
        }

        // Method getter untuk mendapatkan nilai setiap atribut pada kelas Tabel
        public function getBaris()
        {
            return $this->baris;
        }

        public function getKolom()
        {
            return $this->kolom;
        }

        // Method untuk menampilkan list dalam bentuk tabel
        function buatBaris($baris, $kolom, $data) {
            $i = 0;
            $j = 0;

            echo "<table border='2' style='border-collapse:collapse;'>";
                for ($i=0; $i < $baris; $i++) { 
                    echo "<tr>";
                    for ($j=0; $j < $kolom; $j++) { 
                        if ($j == 6) {
                            echo "<td>";
                            echo "<center>" . $data[$i][$j] . "</center>";
                            echo "<td>";
                        }
                        else {
                            echo "<td>";
                            echo $data[$i][$j];
                            echo "<td>";
                        }
                    }
                    echo "<br>";
                    echo "</tr>";
                }
            echo "</table>";
        }        

        // Destructor
        function __destruct()
        {
            
        }
    }
?>